src/main/webapp/js/index.js

```diff
@@ -12,8 +12,92 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/* Date range picker handler:
- * https://www.daterangepicker.com/
+
+/** Fetches receipts from the server and adds them to the DOM. */
+async function searchReceipts() {
+  const label = document.getElementById('search-input').value;
+  const response = await fetch(`/search-receipts?label=${label}`);
+  const receipts = await response.json();
+
+  clearExistingDisplay();
+  displayReceipts(label, receipts);
+}
+
+/**
+ * Clears out receipts display including old receipts and error messages.
+ */
+function clearExistingDisplay() {
+  const elements = document.querySelectorAll(
+      '.col-md-6, .col-md-12.text-center.error-message');
+  elements.forEach(function(element) {
+    element.parentNode.removeChild(element);
+  });
+}
+
+/**
+ * Populates receipt display with newly queried receipts.
+ * @param {string} label User-entered label.
+ * @param {JSON Object} receipts Receipts returned from search query.
+ */
+function displayReceipts(label, receipts) {
+  // If no receipts returned, display an error message. Else, display receipts.
+  if (Object.keys(receipts).length == 0) {
+    createErrorMessageElement(label);
+  } else {
+    receipts.forEach((receipt) => {
+      createReceiptCardElement(receipt);
+    });
+  }
+}
+
+/**
+ * Creates receipt card based on existing HTML template.
+ * This card displays transaction date, store name, trasaction total,
+ * categories, receipt photo, and view/edit/delete buttons.
+ * @param {Receipt} receipt A Receipt object.
+ */
+function createReceiptCardElement(receipt) {
+  const categories = Array.from(receipt.categories);
+
+  // Clone receipt card from template.
+  const receiptCardClone =
+      document.querySelector('#receipt-card-template').content.cloneNode(true);
+
+  // Fill in template fields with correct information.
+  receiptCardClone.querySelector('#timestamp').innerText = receipt.timestamp;
+  receiptCardClone.querySelector('#store-name').innerText = receipt.store;
+  receiptCardClone.querySelector('#total').innerText =
+      'Total: $' + receipt.price;
+  receiptCardClone.querySelector('#c1').innerText = categories[0];
+  receiptCardClone.querySelector('#c2').innerText = categories[1];
+  receiptCardClone.querySelector('#c3').innerText = categories[2];
```

Comment:
Presumably there may be less than 3 categories, you might want to loop through the categories using  `receiptCardClone.querySelector('#c' + i).innerText = ...` up to the 3 categories.